elastic/apm-server
# Add additional config remapping for "api_key" and "secret_token" for libbeat instrumentation wrapper

## Motivation/summary

This PR addresses the [issue](https://github.com/elastic/apm-server/issues/13690) by updating existing wrapper around libbeat instrumentation to additionally propagate erased tracer config keys "api_key" and "secret_token" from elastic agent via environmental variables.

## Checklist


For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Follow instructions in `Building an Elastic Agent container image with a locally built APM Server`  and `Running an Elastic Agent container with a locally built APM Server` section in `TESTING.md`, update the relevant `agent.monitoring` section with "api_key" or "secret_token" in `elastic-agent.yml` to verify that tracing authentication works correctly.

## Related issues

https://github.com/elastic/apm-server/issues/13690
https://github.com/elastic/apm-server/issues/13597
